src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,201 @@
+package duke.util;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import duke.command.AddTaskCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.exception.DukeDeadlineMissingDateException;
+import duke.exception.DukeEventMissingDateException;
+import duke.exception.DukeException;
+import duke.exception.DukeInvalidIndexException;
+import duke.exception.DukeMissingDescriptionException;
+import duke.exception.DukeMissingIndexException;
+import duke.exception.DukeNoSuchCommandException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+/**
+ * This class encapsulates the parser which interprets user input.
+ *
+ * @author Teo Sin Yee
+ */
+public class Parser {
+
+    /**
+     * Parses the raw user input and returns the corresponding Command to execute.
+     *
+     * @param input Raw user input.
+     * @return The corresponding Command.
+     * @throws DukeException If user input is unexpected.
+     */
+    public Command parseRawInput(String input) throws DukeException {
+        String[] splitInput = input.split("\\s+");
+        String commandWord = splitInput[0];
+        switch (commandWord) {
```

Comment:
Perhaps you could make the input case-insensitive. 
```suggestion
        switch (commandWord.toLowerCase()) {
```
So instead of throwing a `DukeNoSuchCommandException()` when, for example, *"List"* is given as input, it can still work as expected.